### DISCUSSION TITLE
Enable Link-Time Optimization (LTO)

### DISCUSSION THREAD

**Discussion by zamazan4ik** [hluk/CopyQ#2931] [Ideas]
Hi!

I noticed that in the CMake files, LTO is not enabled for CopyQ. Enabling this option allows to a compiler perform more aggressive optimizations which usually lead to a smaller binary and more performant/energy efficient application (since LTO allows to perform more CPU-related optimizations - however I don't think that performance question is too critical for the app). Enabling it in the upstream makes it easier to enable LTO for CopyQ maintainers in OS and allows users who get CopyQ from GitHub Releases to get a more efficient binary.

I have made quick tests with the latest CopyQ version + Fedora 41 + GCC 14.2.1 with the `cmake -DCMAKE_BUILD_TYPE=Release -G Ninja . && ninja`. The default Release binary size is 4.5 Mib, Release + LTO (I have added `-flto` to `CMAKE_CXX_FLAGS` and `CMAKE_EXE_LINKER_FLAGS`) is 4 Mib.

Thank you.

**Comment by hluk** [comment]
This seems like an issue for the specific Fedora distribution package. I do not maintain the Fedora package itself.

**Reply by zamazan4ik** [reply]
I built the local version and tested it locally - I didn't install CopyQ from Fedora repos.

If we enable LTO on the CMake level directly in CopyQ, LTO will be automatically enabled for all distros, not only Fedora.
